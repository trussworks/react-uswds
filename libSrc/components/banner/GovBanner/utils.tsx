import React from 'react'
import Icon from '../../Icon/Icon.js'

export type Language = 'english' | 'spanish'

export type TLD = '.gov' | '.mil'

export interface GovBannerCopy {
  header: string
  ariaLabel: string
  headerAction: string
  tldSectionHeader: string
  tldSectionContent: JSX.Element
  httpsSectionHeader: string
  httpsSectionContent: JSX.Element
}

export const getCopy = (language: Language, tld: TLD): GovBannerCopy => {
  switch (language) {
    case 'english':
      return {
        header: 'An official website of the United States government',
        ariaLabel: 'Official website of the United States government',
        headerAction: 'Here’s how you know',
        tldSectionHeader: `Official websites use ${tld}`,
        tldSectionContent: ((): JSX.Element => {
          switch (tld) {
            case '.gov':
              return (
                <>
                  A <strong>.gov</strong> website belongs to an official
                  government organization in the United States.
                </>
              )
            case '.mil':
              return (
                <>
                  A <strong>.mil</strong> website belongs to an official U.S.
                  Department of Defense organization.
                </>
              )
          }
        })(),
        httpsSectionHeader: `Secure ${tld} websites use HTTPS`,
        httpsSectionContent: (
          <>
            A{' '}
            <strong>
              lock (<Icon.Lock aria-label="Locked padlock icon" />)
            </strong>{' '}
            or <strong>https://</strong> means you’ve safely connected to the{' '}
            {tld} website. Share sensitive information only on official, secure
            websites.
          </>
        ),
      }
    case 'spanish':
      return {
        header: 'Un sitio oficial del Gobierno de Estados Unidos',
        ariaLabel: 'Un sitio oficial del Gobierno de Estados Unidos',
        headerAction: 'Así es como usted puede verificarlo',
        tldSectionHeader: `Los sitios web oficiales usan ${tld}`,
        tldSectionContent: ((): JSX.Element => {
          switch (tld) {
            case '.gov':
              return (
                <>
                  Un sitio web <strong>.gov</strong> pertenece a una
                  organización oficial del Gobierno de Estados Unidos.
                </>
              )
            case '.mil':
              return (
                <>
                  Un sitio web <strong>.mil</strong> pertenece a una
                  organización oficial del Departamento de Defensa de EE. UU.
                </>
              )
          }
        })(),
        httpsSectionHeader: `Los sitios web seguros ${tld} usan HTTPS`,
        httpsSectionContent: (
          <>
            Un{' '}
            <strong>
              candado (<Icon.Lock aria-label="Icono de candado cerrado" />)
            </strong>{' '}
            o <strong>https://</strong> significa que usted se conectó de forma
            segura a un sitio web {tld}. Comparta información sensible sólo en
            sitios web oficiales y seguros.
          </>
        ),
      }
  }
}
