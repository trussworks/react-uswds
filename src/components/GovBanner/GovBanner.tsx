import React, { useState } from 'react'
import classnames from 'classnames'

// assets
import flagImg from 'uswds/src/img/us_flag_small.png'
import dotGovIcon from 'uswds/src/img/icon-dot-gov.svg'
import httpsIcon from 'uswds/src/img/icon-https.svg'
import lockIcon from 'uswds/src/img/lock.svg'

type Language = 'english' | 'spanish'

type TLD = '.gov' | '.mil'

interface GovBannerCopy {
  header: string
  headerAction: string
  tldSectionHeader: string
  tldSectionContent: JSX.Element
  httpsSectionHeader: string
  httpsSectionContent: JSX.Element
}

const getCopy = (language: Language, tld: TLD): GovBannerCopy => {
  const lock = (
    <span className="icon-lock">
      {/* eslint-disable-next-line jsx-a11y/no-redundant-roles */}
      <img
        src={lockIcon}
        className="usa-banner__lock-image"
        alt="lock"
        title="Lock"
        role="img"
      />
    </span>
  )

  switch (language) {
    case 'english':
      return {
        header: 'An official website of the United States government',
        headerAction: 'Here’s how you know',
        tldSectionHeader: `Official websites use ${tld}`,
        tldSectionContent: ((): JSX.Element => {
          switch (tld) {
            case '.gov':
              return (
                <>
                  A <strong>.gov</strong> website belongs to an official
                  government organization in the United States.
                </>
              )
            case '.mil':
              return (
                <>
                  A <strong>.mil</strong> website belongs to an official U.S.
                  Department of Defense organization.
                </>
              )
          }
        })(),
        httpsSectionHeader: `Secure ${tld} websites use HTTPS`,
        httpsSectionContent: (
          <>
            A <strong>lock ({lock})</strong> or <strong>https://</strong> means
            you’ve safely connected to the {tld} website. Share sensitive
            information only on official, secure websites.
          </>
        ),
      }
    case 'spanish':
      return {
        header: 'Un sitio oficial del Gobierno de Estados Unidos',
        headerAction: 'Así es como usted puede verificarlo',
        tldSectionHeader: `Los sitios web oficiales usan ${tld}`,
        tldSectionContent: ((): JSX.Element => {
          switch (tld) {
            case '.gov':
              return (
                <>
                  Un sitio web <strong>.gov</strong> pertenece a una
                  organización oficial del Gobierno de Estados Unidos.
                </>
              )
            case '.mil':
              return (
                <>
                  Un sitio web <strong>.mil</strong> pertenece a una
                  organización oficial del Departamento de Defensa de EE. UU.
                </>
              )
          }
        })(),
        httpsSectionHeader: `Los sitios web seguros ${tld} usan HTTPS`,
        httpsSectionContent: (
          <>
            Un <strong>candado ({lock})</strong> o <strong>https://</strong>{' '}
            significa que usted se conectó de forma segura a un sitio web {tld}.
            Comparta información sensible sólo en sitios web oficiales y
            seguros.
          </>
        ),
      }
  }
}

interface GovBannerProps {
  tld?: TLD
  language?: Language
}

export const GovBanner = ({
  tld = '.gov',
  language = 'english',
  className,
  ...sectionProps
}: GovBannerProps & JSX.IntrinsicElements['section']): React.ReactElement => {
  const [isOpen, setOpenState] = useState(false)

  const classes = classnames('usa-banner', className)

  const copy = getCopy(language, tld)

  return (
    <section className={classes} data-testid="govBanner" {...sectionProps}>
      <div className="usa-accordion">
        <header className="usa-banner__header">
          <div className="usa-banner__inner">
            <div className="grid-col-auto">
              <img
                className="usa-banner__header-flag"
                src={flagImg}
                alt="U.S. flag"
              />
            </div>
            <div className="grid-col-fill tablet:grid-col-auto">
              <p className="usa-banner__header-text">{copy.header}</p>
              <p className="usa-banner__header-action" aria-hidden="true">
                {copy.headerAction}
              </p>
            </div>
            <button
              type="button"
              className="usa-accordion__button usa-banner__button"
              aria-expanded={isOpen}
              aria-controls="gov-banner"
              onClick={(): void => {
                setOpenState((isOpen) => !isOpen)
              }}>
              <span className="usa-banner__button-text">
                {copy.headerAction}
              </span>
            </button>
          </div>
        </header>
        <div
          className="usa-banner__content usa-accordion__content"
          id="gov-banner"
          hidden={!isOpen}>
          <div className="grid-row grid-gap-lg">
            <div className="usa-banner__guidance tablet:grid-col-6">
              <img
                className="usa-banner__icon usa-media-block__img"
                src={dotGovIcon}
                role="img"
                alt=""
                aria-hidden="true"
              />
              <div className="usa-media-block__body">
                <p>
                  <strong>{copy.tldSectionHeader}</strong>
                  <br />
                  {copy.tldSectionContent}
                </p>
              </div>
            </div>
            <div className="usa-banner__guidance tablet:grid-col-6">
              <img
                className="usa-banner__icon usa-media-block__img"
                src={httpsIcon}
                role="img"
                alt=""
                aria-hidden="true"
              />
              <div className="usa-media-block__body">
                <p>
                  <strong>{copy.httpsSectionHeader}</strong>
                  <br />
                  {copy.httpsSectionContent}
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  )
}

export default GovBanner
