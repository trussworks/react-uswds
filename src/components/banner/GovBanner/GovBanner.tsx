import React, { ReactElement, useState, JSX } from 'react'
import { Banner } from '../Banner/Banner'
import { BannerHeader } from '../BannerHeader/BannerHeader'
import { BannerFlag } from '../BannerFlag/BannerFlag'
import { BannerButton } from '../BannerButton/BannerButton'
import { BannerContent } from '../BannerContent/BannerContent'
import { BannerGuidance } from '../BannerGuidance/BannerGuidance'
import { BannerIcon } from '../BannerIcon/BannerIcon'
import { MediaBlockBody } from '../../mediablock/MediaBlockBody/MediaBlockBody'
import { Icon } from '../../Icon/Icons'

// assets
import flagImg from '@uswds/uswds/img/us_flag_small.png'
import dotGovIcon from '@uswds/uswds/img/icon-dot-gov.svg'
import httpsIcon from '@uswds/uswds/img/icon-https.svg'

type Language = 'english' | 'spanish'

type TLD = '.gov' | '.mil'

interface GovBannerCopy {
  header: string
  ariaLabel: string
  headerAction: string
  tldSectionHeader: string
  tldSectionContent: JSX.Element
  httpsSectionHeader: string
  httpsSectionContent: JSX.Element
}

const getCopy = (language: Language, tld: TLD): GovBannerCopy => {
  switch (language) {
    case 'english':
      return {
        header: 'An official website of the United States government',
        ariaLabel: 'Official website of the United States government',
        headerAction: 'Here’s how you know',
        tldSectionHeader: `Official websites use ${tld}`,
        tldSectionContent: ((): JSX.Element => {
          switch (tld) {
            case '.gov':
              return (
                <>
                  A <strong>.gov</strong> website belongs to an official
                  government organization in the United States.
                </>
              )
            case '.mil':
              return (
                <>
                  A <strong>.mil</strong> website belongs to an official U.S.
                  Department of Defense organization.
                </>
              )
          }
        })(),
        httpsSectionHeader: `Secure ${tld} websites use HTTPS`,
        httpsSectionContent: (
          <>
            A{' '}
            <strong>
              lock (<Icon.Lock aria-label="Locked padlock icon" />)
            </strong>{' '}
            or <strong>https://</strong> means you’ve safely connected to the{' '}
            {tld} website. Share sensitive information only on official, secure
            websites.
          </>
        ),
      }
    case 'spanish':
      return {
        header: 'Un sitio oficial del Gobierno de Estados Unidos',
        ariaLabel: 'Un sitio oficial del Gobierno de Estados Unidos',
        headerAction: 'Así es como usted puede verificarlo',
        tldSectionHeader: `Los sitios web oficiales usan ${tld}`,
        tldSectionContent: ((): JSX.Element => {
          switch (tld) {
            case '.gov':
              return (
                <>
                  Un sitio web <strong>.gov</strong> pertenece a una
                  organización oficial del Gobierno de Estados Unidos.
                </>
              )
            case '.mil':
              return (
                <>
                  Un sitio web <strong>.mil</strong> pertenece a una
                  organización oficial del Departamento de Defensa de EE. UU.
                </>
              )
          }
        })(),
        httpsSectionHeader: `Los sitios web seguros ${tld} usan HTTPS`,
        httpsSectionContent: (
          <>
            Un{' '}
            <strong>
              candado (<Icon.Lock aria-label="Icono de candado cerrado" />)
            </strong>{' '}
            o <strong>https://</strong> significa que usted se conectó de forma
            segura a un sitio web {tld}. Comparta información sensible sólo en
            sitios web oficiales y seguros.
          </>
        ),
      }
  }
}

type GovBannerProps = {
  tld?: TLD
  language?: Language
}

export const GovBanner = ({
  tld = '.gov',
  language = 'english',
  className,
  ...sectionProps
}: GovBannerProps & JSX.IntrinsicElements['section']): ReactElement => {
  const [isOpen, setIsOpen] = useState(false)

  const {
    header,
    ariaLabel,
    headerAction,
    httpsSectionHeader,
    httpsSectionContent,
    tldSectionHeader,
    tldSectionContent,
  } = getCopy(language, tld)

  return (
    <Banner
      className={className}
      data-testid="govBanner"
      aria-label={ariaLabel}
      {...sectionProps}>
      <BannerHeader
        isOpen={isOpen}
        flagImg={<BannerFlag src={flagImg} aria-hidden alt="" />}
        headerText={header}
        headerActionText={headerAction}>
        <BannerButton
          isOpen={isOpen}
          aria-controls="gov-banner"
          onClick={(): void => {
            setIsOpen((previousIsOpen) => !previousIsOpen)
          }}>
          {headerAction}
        </BannerButton>
      </BannerHeader>
      <BannerContent id="gov-banner" isOpen={isOpen}>
        <div className="grid-row grid-gap-lg">
          <BannerGuidance className="tablet:grid-col-6">
            <BannerIcon src={dotGovIcon} alt="" />
            <MediaBlockBody>
              <p>
                <strong>{tldSectionHeader}</strong>
                <br />
                {tldSectionContent}
              </p>
            </MediaBlockBody>
          </BannerGuidance>
          <BannerGuidance className="tablet:grid-col-6">
            <BannerIcon src={httpsIcon} alt="" />
            <MediaBlockBody>
              <p>
                <strong>{httpsSectionHeader}</strong>
                <br />
                {httpsSectionContent}
              </p>
            </MediaBlockBody>
          </BannerGuidance>
        </div>
      </BannerContent>
    </Banner>
  )
}

export default GovBanner
