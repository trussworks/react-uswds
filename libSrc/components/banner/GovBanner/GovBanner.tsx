import React, { type ReactElement, useState } from 'react'
import Banner, { type BannerProps } from '../Banner/Banner.js'
import BannerHeader from '../BannerHeader/BannerHeader.js'
import BannerFlag from '../BannerFlag/BannerFlag.js'
import BannerButton from '../BannerButton/BannerButton.js'
import BannerContent from '../BannerContent/BannerContent.js'
import BannerGuidance from '../BannerGuidance/BannerGuidance.js'
import BannerIcon from '../BannerIcon/BannerIcon.js'
import MediaBlockBody from '../../mediablock/MediaBlockBody/MediaBlockBody.js'
import { type Language, type TLD, getCopy } from './utils.js'
import Grid from '../../grid/Grid/Grid.js'

// assets
import flagImg from '@uswds/uswds/img/us_flag_small.png'
import dotGovIcon from '@uswds/uswds/img/icon-dot-gov.svg'
import httpsIcon from '@uswds/uswds/img/icon-https.svg'

export interface BaseGovBannerProps {
  tld?: TLD
  language?: Language
}

export type GovBannerProps = BaseGovBannerProps & BannerProps

const GovBanner = ({
  tld = '.gov',
  language = 'english',
  className,
  ...sectionProps
}: GovBannerProps): ReactElement => {
  const [isOpen, setIsOpen] = useState(false)

  const {
    header,
    ariaLabel,
    headerAction,
    httpsSectionHeader,
    httpsSectionContent,
    tldSectionHeader,
    tldSectionContent,
  } = getCopy(language, tld)

  return (
    <Banner
      className={className}
      data-testid="govBanner"
      aria-label={ariaLabel}
      {...sectionProps}>
      <BannerHeader
        isOpen={isOpen}
        flagImg={<BannerFlag src={flagImg} aria-hidden alt="" />}
        headerText={header}
        headerActionText={headerAction}>
        <BannerButton
          isOpen={isOpen}
          aria-controls="gov-banner"
          onClick={(): void => {
            setIsOpen((previousIsOpen) => !previousIsOpen)
          }}>
          {headerAction}
        </BannerButton>
      </BannerHeader>
      <BannerContent id="gov-banner" isOpen={isOpen}>
        <Grid row gap="lg">
          <BannerGuidance className="tablet:grid-col-6">
            <BannerIcon src={dotGovIcon} alt="" />
            <MediaBlockBody>
              <p>
                <strong>{tldSectionHeader}</strong>
                <br />
                {tldSectionContent}
              </p>
            </MediaBlockBody>
          </BannerGuidance>
          <BannerGuidance className="tablet:grid-col-6">
            <BannerIcon src={httpsIcon} alt="" />
            <MediaBlockBody>
              <p>
                <strong>{httpsSectionHeader}</strong>
                <br />
                {httpsSectionContent}
              </p>
            </MediaBlockBody>
          </BannerGuidance>
        </Grid>
      </BannerContent>
    </Banner>
  )
}

export default GovBanner
