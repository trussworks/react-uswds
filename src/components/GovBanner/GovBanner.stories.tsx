import React from 'react'
import { GovBanner } from './GovBanner'

export default {
  title: 'Components/Banner',
  component: GovBanner,
  parameters: {
    docs: {
      description: {
        component: `
### USWDS 2.0 Banner component

Source: https://designsystem.digital.gov/components/banner/
`,
      },
    },
  },
}

export const govBannerDefault = (): React.ReactElement => (
  <GovBanner aria-label="Official government website" />
)

export const govBannerEnglishDotGov = (): React.ReactElement => (
  <GovBanner
    language="english"
    tld=".gov"
    aria-label="Official government website"
  />
)

export const govBannerEnglishDotMil = (): React.ReactElement => (
  <GovBanner
    language="english"
    tld=".mil"
    aria-label="Official government website"
  />
)

export const govBannerSpanishDotGov = (): React.ReactElement => (
  <GovBanner
    language="spanish"
    tld=".gov"
    aria-label="Official government website"
  />
)

export const govBannerSpanishDotMil = (): React.ReactElement => (
  <GovBanner
    language="spanish"
    tld=".mil"
    aria-label="Official government website"
  />
)
