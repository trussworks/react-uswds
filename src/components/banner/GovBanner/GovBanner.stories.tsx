import React from 'react'
import { GovBanner } from './GovBanner'

export default {
  title: 'Components/Banner',
  component: GovBanner,
  parameters: {
    docs: {
      description: {
        component: `
### USWDS 3.0 Banner component

Source: https://designsystem.digital.gov/components/banner/
`,
      },
    },
  },
}

export const GovBannerDefault = (): React.ReactElement => (
  <GovBanner aria-label="Official government website" />
)

export const GovBannerEnglishDotGov = (): React.ReactElement => (
  <GovBanner
    language="english"
    tld=".gov"
    aria-label="Official government website"
  />
)

export const GovBannerEnglishDotMil = (): React.ReactElement => (
  <GovBanner
    language="english"
    tld=".mil"
    aria-label="Official government website"
  />
)

export const GovBannerSpanishDotGov = (): React.ReactElement => (
  <GovBanner
    language="spanish"
    tld=".gov"
    aria-label="Official government website"
  />
)

export const GovBannerSpanishDotMil = (): React.ReactElement => (
  <GovBanner
    language="spanish"
    tld=".mil"
    aria-label="Official government website"
  />
)
