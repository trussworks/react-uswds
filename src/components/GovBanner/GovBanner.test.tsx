import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'

import { GovBanner } from './GovBanner'

describe('GovBanner component', () => {
  it('renders without errors', () => {
    const { queryByTestId } = render(<GovBanner />)
    expect(queryByTestId('govBanner')).toBeInTheDocument()
  })

  it('renders section attributes passed in through props', () => {
    const { queryByTestId } = render(
      <GovBanner aria-label="Official government website" />
    )
    expect(queryByTestId('govBanner')).toHaveAttribute('aria-label')
  })

  it('renders with language and tld props passed', () => {
    const { queryByTestId } = render(
      <GovBanner language="english" tld=".mil" />
    )
    expect(queryByTestId('govBanner')).toBeInTheDocument
  })

  describe('static content', () => {
    it('renders consistently with default props', () => {
      const tree = renderer.create(<GovBanner />).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders consistently in English for .gov sites', () => {
      const tree = renderer
        .create(<GovBanner language="english" tld=".gov" />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders consistently in English for .mil sites', () => {
      const tree = renderer
        .create(<GovBanner language="english" tld=".mil" />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders consistently in Spanish for .gov sites', () => {
      const tree = renderer
        .create(<GovBanner language="spanish" tld=".gov" />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders consistently in Spanish for .mil sites', () => {
      const tree = renderer
        .create(<GovBanner language="spanish" tld=".mil" />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
