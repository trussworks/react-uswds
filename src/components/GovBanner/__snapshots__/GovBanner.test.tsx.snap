// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GovBanner component static content renders consistently in English for .gov sites 1`] = `
<section
  className="usa-banner"
  data-testid="govBanner"
>
  <div
    className="usa-accordion"
  >
    <header
      className="usa-banner__header"
    >
      <div
        className="usa-banner__inner"
      >
        <div
          className="grid-col-auto"
        >
          <img
            alt="U.S. flag"
            className="usa-banner__header-flag"
            src="test-file-stub"
          />
        </div>
        <div
          className="grid-col-fill tablet:grid-col-auto"
        >
          <p
            className="usa-banner__header-text"
          >
            An official website of the United States government
          </p>
          <p
            aria-hidden="true"
            className="usa-banner__header-action"
          >
            Here’s how you know
          </p>
        </div>
        <button
          aria-controls="gov-banner"
          aria-expanded={false}
          className="usa-accordion__button usa-banner__button"
          onClick={[Function]}
          type="button"
        >
          <span
            className="usa-banner__button-text"
          >
            Here’s how you know
          </span>
        </button>
      </div>
    </header>
    <div
      className="usa-banner__content usa-accordion__content"
      hidden={true}
      id="gov-banner"
    >
      <div
        className="grid-row grid-gap-lg"
      >
        <div
          className="usa-banner__guidance tablet:grid-col-6"
        >
          <img
            alt=""
            aria-hidden="true"
            className="usa-banner__icon usa-media-block__img"
            role="img"
            src="test-file-stub"
          />
          <div
            className="usa-media-block__body"
          >
            <p>
              <strong>
                Official websites use .gov
              </strong>
              <br />
              A 
              <strong>
                .gov
              </strong>
               website belongs to an official government organization in the United States.
            </p>
          </div>
        </div>
        <div
          className="usa-banner__guidance tablet:grid-col-6"
        >
          <img
            alt=""
            aria-hidden="true"
            className="usa-banner__icon usa-media-block__img"
            role="img"
            src="test-file-stub"
          />
          <div
            className="usa-media-block__body"
          >
            <p>
              <strong>
                Secure .gov websites use HTTPS
              </strong>
              <br />
              A 
              <strong>
                lock (
                <span
                  className="icon-lock"
                >
                  <img
                    alt="lock"
                    className="usa-banner__lock-image"
                    role="img"
                    src="test-file-stub"
                    title="Lock"
                  />
                </span>
                )
              </strong>
               or 
              <strong>
                https://
              </strong>
               means you’ve safely connected to the 
              .gov
               website. Share sensitive information only on official, secure websites.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`GovBanner component static content renders consistently in English for .mil sites 1`] = `
<section
  className="usa-banner"
  data-testid="govBanner"
>
  <div
    className="usa-accordion"
  >
    <header
      className="usa-banner__header"
    >
      <div
        className="usa-banner__inner"
      >
        <div
          className="grid-col-auto"
        >
          <img
            alt="U.S. flag"
            className="usa-banner__header-flag"
            src="test-file-stub"
          />
        </div>
        <div
          className="grid-col-fill tablet:grid-col-auto"
        >
          <p
            className="usa-banner__header-text"
          >
            An official website of the United States government
          </p>
          <p
            aria-hidden="true"
            className="usa-banner__header-action"
          >
            Here’s how you know
          </p>
        </div>
        <button
          aria-controls="gov-banner"
          aria-expanded={false}
          className="usa-accordion__button usa-banner__button"
          onClick={[Function]}
          type="button"
        >
          <span
            className="usa-banner__button-text"
          >
            Here’s how you know
          </span>
        </button>
      </div>
    </header>
    <div
      className="usa-banner__content usa-accordion__content"
      hidden={true}
      id="gov-banner"
    >
      <div
        className="grid-row grid-gap-lg"
      >
        <div
          className="usa-banner__guidance tablet:grid-col-6"
        >
          <img
            alt=""
            aria-hidden="true"
            className="usa-banner__icon usa-media-block__img"
            role="img"
            src="test-file-stub"
          />
          <div
            className="usa-media-block__body"
          >
            <p>
              <strong>
                Official websites use .mil
              </strong>
              <br />
              A 
              <strong>
                .mil
              </strong>
               website belongs to an official U.S. Department of Defense organization.
            </p>
          </div>
        </div>
        <div
          className="usa-banner__guidance tablet:grid-col-6"
        >
          <img
            alt=""
            aria-hidden="true"
            className="usa-banner__icon usa-media-block__img"
            role="img"
            src="test-file-stub"
          />
          <div
            className="usa-media-block__body"
          >
            <p>
              <strong>
                Secure .mil websites use HTTPS
              </strong>
              <br />
              A 
              <strong>
                lock (
                <span
                  className="icon-lock"
                >
                  <img
                    alt="lock"
                    className="usa-banner__lock-image"
                    role="img"
                    src="test-file-stub"
                    title="Lock"
                  />
                </span>
                )
              </strong>
               or 
              <strong>
                https://
              </strong>
               means you’ve safely connected to the 
              .mil
               website. Share sensitive information only on official, secure websites.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`GovBanner component static content renders consistently in Spanish for .gov sites 1`] = `
<section
  className="usa-banner"
  data-testid="govBanner"
>
  <div
    className="usa-accordion"
  >
    <header
      className="usa-banner__header"
    >
      <div
        className="usa-banner__inner"
      >
        <div
          className="grid-col-auto"
        >
          <img
            alt="U.S. flag"
            className="usa-banner__header-flag"
            src="test-file-stub"
          />
        </div>
        <div
          className="grid-col-fill tablet:grid-col-auto"
        >
          <p
            className="usa-banner__header-text"
          >
            Un sitio oficial del Gobierno de Estados Unidos
          </p>
          <p
            aria-hidden="true"
            className="usa-banner__header-action"
          >
            Así es como usted puede verificarlo
          </p>
        </div>
        <button
          aria-controls="gov-banner"
          aria-expanded={false}
          className="usa-accordion__button usa-banner__button"
          onClick={[Function]}
          type="button"
        >
          <span
            className="usa-banner__button-text"
          >
            Así es como usted puede verificarlo
          </span>
        </button>
      </div>
    </header>
    <div
      className="usa-banner__content usa-accordion__content"
      hidden={true}
      id="gov-banner"
    >
      <div
        className="grid-row grid-gap-lg"
      >
        <div
          className="usa-banner__guidance tablet:grid-col-6"
        >
          <img
            alt=""
            aria-hidden="true"
            className="usa-banner__icon usa-media-block__img"
            role="img"
            src="test-file-stub"
          />
          <div
            className="usa-media-block__body"
          >
            <p>
              <strong>
                Los sitios web oficiales usan .gov
              </strong>
              <br />
              Un sitio web 
              <strong>
                .gov
              </strong>
               pertenece a una organización oficial del Gobierno de Estados Unidos.
            </p>
          </div>
        </div>
        <div
          className="usa-banner__guidance tablet:grid-col-6"
        >
          <img
            alt=""
            aria-hidden="true"
            className="usa-banner__icon usa-media-block__img"
            role="img"
            src="test-file-stub"
          />
          <div
            className="usa-media-block__body"
          >
            <p>
              <strong>
                Los sitios web seguros .gov usan HTTPS
              </strong>
              <br />
              Un 
              <strong>
                candado (
                <span
                  className="icon-lock"
                >
                  <img
                    alt="lock"
                    className="usa-banner__lock-image"
                    role="img"
                    src="test-file-stub"
                    title="Lock"
                  />
                </span>
                )
              </strong>
               o 
              <strong>
                https://
              </strong>
               
              significa que usted se conectó de forma segura a un sitio web 
              .gov
              . Comparta información sensible sólo en sitios web oficiales y seguros.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`GovBanner component static content renders consistently in Spanish for .mil sites 1`] = `
<section
  className="usa-banner"
  data-testid="govBanner"
>
  <div
    className="usa-accordion"
  >
    <header
      className="usa-banner__header"
    >
      <div
        className="usa-banner__inner"
      >
        <div
          className="grid-col-auto"
        >
          <img
            alt="U.S. flag"
            className="usa-banner__header-flag"
            src="test-file-stub"
          />
        </div>
        <div
          className="grid-col-fill tablet:grid-col-auto"
        >
          <p
            className="usa-banner__header-text"
          >
            Un sitio oficial del Gobierno de Estados Unidos
          </p>
          <p
            aria-hidden="true"
            className="usa-banner__header-action"
          >
            Así es como usted puede verificarlo
          </p>
        </div>
        <button
          aria-controls="gov-banner"
          aria-expanded={false}
          className="usa-accordion__button usa-banner__button"
          onClick={[Function]}
          type="button"
        >
          <span
            className="usa-banner__button-text"
          >
            Así es como usted puede verificarlo
          </span>
        </button>
      </div>
    </header>
    <div
      className="usa-banner__content usa-accordion__content"
      hidden={true}
      id="gov-banner"
    >
      <div
        className="grid-row grid-gap-lg"
      >
        <div
          className="usa-banner__guidance tablet:grid-col-6"
        >
          <img
            alt=""
            aria-hidden="true"
            className="usa-banner__icon usa-media-block__img"
            role="img"
            src="test-file-stub"
          />
          <div
            className="usa-media-block__body"
          >
            <p>
              <strong>
                Los sitios web oficiales usan .mil
              </strong>
              <br />
              Un sitio web 
              <strong>
                .mil
              </strong>
               pertenece a una organización oficial del Departamento de Defensa de EE. UU.
            </p>
          </div>
        </div>
        <div
          className="usa-banner__guidance tablet:grid-col-6"
        >
          <img
            alt=""
            aria-hidden="true"
            className="usa-banner__icon usa-media-block__img"
            role="img"
            src="test-file-stub"
          />
          <div
            className="usa-media-block__body"
          >
            <p>
              <strong>
                Los sitios web seguros .mil usan HTTPS
              </strong>
              <br />
              Un 
              <strong>
                candado (
                <span
                  className="icon-lock"
                >
                  <img
                    alt="lock"
                    className="usa-banner__lock-image"
                    role="img"
                    src="test-file-stub"
                    title="Lock"
                  />
                </span>
                )
              </strong>
               o 
              <strong>
                https://
              </strong>
               
              significa que usted se conectó de forma segura a un sitio web 
              .mil
              . Comparta información sensible sólo en sitios web oficiales y seguros.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`GovBanner component static content renders consistently with default props 1`] = `
<section
  className="usa-banner"
  data-testid="govBanner"
>
  <div
    className="usa-accordion"
  >
    <header
      className="usa-banner__header"
    >
      <div
        className="usa-banner__inner"
      >
        <div
          className="grid-col-auto"
        >
          <img
            alt="U.S. flag"
            className="usa-banner__header-flag"
            src="test-file-stub"
          />
        </div>
        <div
          className="grid-col-fill tablet:grid-col-auto"
        >
          <p
            className="usa-banner__header-text"
          >
            An official website of the United States government
          </p>
          <p
            aria-hidden="true"
            className="usa-banner__header-action"
          >
            Here’s how you know
          </p>
        </div>
        <button
          aria-controls="gov-banner"
          aria-expanded={false}
          className="usa-accordion__button usa-banner__button"
          onClick={[Function]}
          type="button"
        >
          <span
            className="usa-banner__button-text"
          >
            Here’s how you know
          </span>
        </button>
      </div>
    </header>
    <div
      className="usa-banner__content usa-accordion__content"
      hidden={true}
      id="gov-banner"
    >
      <div
        className="grid-row grid-gap-lg"
      >
        <div
          className="usa-banner__guidance tablet:grid-col-6"
        >
          <img
            alt=""
            aria-hidden="true"
            className="usa-banner__icon usa-media-block__img"
            role="img"
            src="test-file-stub"
          />
          <div
            className="usa-media-block__body"
          >
            <p>
              <strong>
                Official websites use .gov
              </strong>
              <br />
              A 
              <strong>
                .gov
              </strong>
               website belongs to an official government organization in the United States.
            </p>
          </div>
        </div>
        <div
          className="usa-banner__guidance tablet:grid-col-6"
        >
          <img
            alt=""
            aria-hidden="true"
            className="usa-banner__icon usa-media-block__img"
            role="img"
            src="test-file-stub"
          />
          <div
            className="usa-media-block__body"
          >
            <p>
              <strong>
                Secure .gov websites use HTTPS
              </strong>
              <br />
              A 
              <strong>
                lock (
                <span
                  className="icon-lock"
                >
                  <img
                    alt="lock"
                    className="usa-banner__lock-image"
                    role="img"
                    src="test-file-stub"
                    title="Lock"
                  />
                </span>
                )
              </strong>
               or 
              <strong>
                https://
              </strong>
               means you’ve safely connected to the 
              .gov
               website. Share sensitive information only on official, secure websites.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
